packages/utils/test/merge.test.ts

```diff
@@ -101,4 +115,29 @@ describe('merge', () => {
             foo: 2
         })
     })
+
+    it('class instances are handled as object references', () => {
+        // eslint-disable-next-line @typescript-eslint/no-extraneous-class
+        class Foo {}
+        const o1 = {
+            foo: 1
+        }
+        const foo = new Foo()
+        const o2 = {
+            foo
+        }
+        expect(merge(o1, o2).foo).toBe(foo)
```

Comment:
I would also check that the values within Foo haven't been affected.
```suggestion
        class Foo {
            values: Record<string, unknown>
        }
        const foo1 = new Foo()
        foo1.values = {
            x: 5,
            y: 6
        }
        const foo2 = new Foo()
        foo2.values = {
            y: 7
            z: 8
        }
        const o1 = {
            foo: foo1
        }
        const o2 = {
            foo: foo2
        }
        const result = merge(o1, o2)
        expect(result.foo).toBe(foo1)
        expect(result.foo.values).toEqual({
            x: 5,
            y: 6
        })
```

Reply:
Good idea. The merging goes other way around: `foo2` should be the value, and values therefore `{ 7, 8 }`